### ISSUE TITLE
Provide a way to handle user session invalidation originated from third party

### ISSUE THREAD

**Issue by VijithaEkanayake** [h2oai/wave#1530]
#### Is your feature request related to a problem? Please describe
We have an OIDC-enabled wave application and it working fine with the usual flow. However, we faced a problem when the user session is terminated in some other way (with another apps logout or validate the session via keycloak), the wave app is not notified even though we register the system.logout event as mentioned in the [docs](https://wave.h2o.ai/docs/routing#handling-user-logout).

When this has happened wave server gives the same old access token and refresh token to the application because of the wave session cookie seems not deleted or expired. So all the API calls failed due to the tokens are not active anymore.

We want to know is there a way to notify the app that the user session has been already invalidated somehow so that the app can redirect the user to the sign-in page for a re-login.

#### Describe the solution you'd like

Provide a way from wave SDK to notify the application via system.logout or some other mechanism that the session has invalidated. So that app can handle it based on the use case.

**Comment by dulajra** [comment]
FYA @mtanco

**Comment by lo5** [comment]
I'd love to see a PR for this based on what you think the ideal solution is.

**Comment by mturoci** [comment]
@VijithaEkanayake-zz can this one be closed since no PR was provided for a year?

**Comment by cpatrickalves** [comment]
I am having a similar issue.
I have a Keycloak that is used by several apps.
If users log out from one app and try to access the Wave app, it gets:

```
 # {"error":"invalid session","session_id":"1dc81a32-d606-4b92-984e-41875488aed8","t":"oauth2_session"}
```

Also, get this error when user tries to open the app: 
```
14:48:21 # {"error":"oauth2: \"unauthorized_client\" \"Invalid client or Invalid client credentials\"","t":"oauth2_exchange"}
```
Which shows in browser:
<img width="270" alt="image" src="https://github.com/h2oai/wave/assets/22003608/d77eb581-bba2-42c8-926f-69557f517a73">

I've tried to debug this in the Python app to handle this, but seems this is handled by the Waved (go). 

Any tips?

**Comment by mturoci** [comment]
> Also, get this error when user tries to open the app:

Yes, that's expected. Waved still has the previous user session stored in a cookie, but there is no corresponding session on the Keycloak side thus the flow fails.

One way you can try is to set a backchannel logout so that Keycloak will notify waved about logout so that session cookie is cleared properly and user taken through full login flow. TBH I would be surprised if our logout endpoint would be ready for this but it's worth a shot.

> I've tried to debug this in the Python app to handle this, but seems this is handled by the Waved (go).

Correct, all the oauth stuff is dealt with in Go.

**Comment by cpatrickalves** [comment]
I'll try, thanks for your help.
the logout endpoint would be` {APP_URL}/_auth/logout` , correct?

**Comment by mturoci** [comment]
Yes.

Another option that comes to my mind is to use a single waved for N apps (I assume your deployment has 1 waved app atm). This was the original idea for running multiple apps. That way, hitting logout in the app should clear the session automatically and give you what you want.
